moveit_experimental/moveit_jog_arm/src/collision_check_thread.cpp

```diff
@@ -113,15 +118,20 @@ void CollisionCheckThread::startMainLoop(JogArmShared& shared_variables)
 
     // If we are far from a collision, velocity_scale should be 1.
     // If we are very close to a collision, velocity_scale should be ~zero.
-    // When collision_proximity_threshold is breached, start decelerating exponentially.
-    else if (collision_distance < parameters_.collision_proximity_threshold)
+    // When scene_collision_proximity_threshold is breached, start decelerating exponentially.
+    else if (scene_collision_distance < parameters_.scene_collision_proximity_threshold)
     {
       // velocity_scale = e ^ k * (collision_distance - threshold)
       // k = - ln(0.001) / collision_proximity_threshold
       // velocity_scale should equal one when collision_distance is at collision_proximity_threshold.
       // velocity_scale should equal 0.001 when collision_distance is at zero.
-      velocity_scale =
-          exp(velocity_scale_coefficient * (collision_distance - parameters_.collision_proximity_threshold));
+      velocity_scale = exp(scene_velocity_scale_coefficient *
+                           (scene_collision_distance - parameters_.scene_collision_proximity_threshold));
+    }
+    else if (self_collision_distance < parameters_.self_collision_proximity_threshold)
```

Comment:
This second else-if could get skipped if the first else-if is true... even if the robot is very close to self-collision. Maybe it would be better to do:

```
else if ((scene_collision_distance < parameters_.scene_collision_proximity_threshold) || (self_collision_distance < parameters_.self_collision_proximity_threshold))
{
  double self_collision_scale = ...
  double scene_collision_scale = ...
  velocity_scale = std::min(self_collision_scale, scene_collision_scale);
}
```

Reply:
Yeah--see my comment above. I did something similar to what you suggested.